### ISSUE TITLE
[BLM] Fire/Blizzard 1 and 2 condition

### ISSUE THREAD

**Issue by kaedys** [daemitus/XIVComboPlugin#67]
Currently, the `Fire 1/3 Feature` and the `Blizzard 1/3 Feature` trigger simple when outside of Astral Fire or Umbral Ice, respective.  However, Fire I and Blizzard I only add a single stack to the Astral/Umbral gauge, while Fire III and Blizzard III automatically set it to 3.  If one has only a single stack on the gauge (for example, when using Transpose, especially when below level 76 and thus lacking Umbral Soul for keeping Enochian rolling between trash pulls), there's not a convenient method of jumping the gauge to 3, and using Fire I or Blizzard I twice to get the stacks is a rather stiff DPS loss.  Can these two features be changed to shift to Fire III and Blizzard III when not in Astral III or Umbral III? Or probably just if at <Astral II/Umbral II, since at 2, Fire I and Blizzard I will stack it to 3 in a single cast.

Similarly, when AoEing, the `Ice 2 Feature` becomes Freeze if you have Umbral Ice at all, leaving no easy way to get Umbral Ice III (and thus the powerful regen and the hasted cast of the first Fire II/High Fire II).

To summarize the request:
* `Fire 1/3 Feature` converts Fire I to Fire III if not in Astral Fire _or if only in Astral Fire I._
* `Blizzard 1/3 Feature` converts Blizzard I to Blizzard III if not in Umbral Ice _or if only in Umbral Ice I._
* `Ice 2 Feature` becomes Freeze _only if in Umbral Ice III._

**Comment by Grammernatzi** [comment]
The thing with this is, if you're stuck in UI1/2, you're not supposed to build Umbral Ice. It's more DPS-effective to use Blizzard 4/Freeze and then use Fire 3/Fire 2. The way you are suggesting is actually *worse* DPS-wise. Same thing if you're in Astral Fire; it is better DPS-wise to sharpcast Fire 1 rather than use Fire 3, and even just Fire 1 spam is better if you don't have Sharpcast available due to the chance for proc. This is even more true when you have Paradox available, as that is one of your highest damaging abilities and will raise AF/UI while not having its damage affected at all by the AF multiplier.

**Comment by kaedys** [comment]
I'm talking well before Paradox is in play.  This is aimed more at the <76 "I don't have Umbral Soul" issue, where you're Transposing back and forth while running between trash pulls.  That leaves you at only UI 1 or UF 1.

I also fail to see how popping Fire 1 twice is superior to Fire 3 once and then into your actual spender, _especially_ if you have Umbral Hearts, since those Fire 1 casts consume the hearts on very low damage spells.  Especially for Umbral Ice, the issue is that if you're only at 1 stack, the mana regen multiplier is low enough that it takes 3-4 ticks, rather than 2, to max out your mana again.  That's a _lot_ of extra time spent in UI, _or_ a shortened Fire phase, which itself is a pretty considerable loss (most likely, you'd either be sacrificing a Flare, _or_ your Flare enhancement, or both).  In Astral Fire, you're sacrificing 25% of your first 3 Fire IV casts (that's ~360 potency loss) _or_ swapping your first 1-2 (depending on procs) Fire IV casts for Fire I (288 potency loss for the first cast, 252 potency loss for the second one, made up by up to about 108 potency if either proc Fire III).  I really don't see how using a Fire 3 in that spot would be worse than just carrying on your rotation as if you _were_ at UF3 or spamming Fire I twice to get there.    It's also worth noting that you only get the hasted spell-cast if you're at UI 3 or UF 3, and since the element-swapping spells all have a >GCD cast time, not having that spell hasting also wastes DPS time.

That said, I don't see how it would block Paradox under normal usage.  This would only apply when you're at UF1 or UI1, and Paradox requires hitting UF3 or UI3.  The only time it would really be applicable is if you're running the style of using Transpose into Paradox and skipping the hasted cast of the opposing element.  And to be honest, I fully expect that particular use-case to be patched out at some point, I doubt their intent is for BLM to be using Transpose rotationally at max level.  And it'd be as easy as making it so you only get Paradox if you swap to the other element by casting an opposing-element spell.

Still, I supposed this could be added as an additional option rather than forcibly baked into the existing one.  ie. The existing `Fire 1/3 Feature` would work as it currently does, but then have an additional `Expanded Fire 1/3 Feature` that also makes it convert to Fire III if at only Astral Fire I.  That would let you simply not select the new option if you're using the Transpose -> Paradox setup.

**Comment by Grammernatzi** [comment]
> I also fail to see how popping Fire 1 twice is superior to Fire 3 once and then into your actual spender,

Because Fire 3 costs more than double the MP. It *destroys* the time you can spend in AF. It's absolutely not worth it, especially if you have Sharpcast. And even without it, fire 1 twice costs less MP and keeps you in AF longer, and has the chance to proc Firestarter, which is more than worth it.

> Especially for Umbral Ice, the issue is that if you're only at 1 stack, the mana regen multiplier is low enough that it takes 3-4 ticks, rather than 2, to max out your mana again.

Blizzard 3 takes more than a tick to cast. And you have to spend time getting Umbral Hearts anyway, as well as likely casting Thunder. Spending as little time in UI is the priority. I could see a use case for this before you get Blizzard 4, but that's it. You should be getting enough MP with Blizzard 4/Freeze and Thunder 2/3/4 otherwise, even at UI1.

Basically, the best solution I could see for this is maybe a feature to just have Blizz 3 level sync into Blizz 1, rather than replace it in UI? That'd make more sense to me, and you could still use Paradox with the Fire 1 button, assuming it overwrites Fire 3 (since you absolutely never want to skip it). Freeze is also useless until 58, so maybe just have the Ice 2 feature not replace it until then.

**Comment by Grammernatzi** [comment]
Okay, so I did some testing with Blizzard: if you transpose from AF to UI, you have to wait a tick to cast anything if you have 0 MP. If you don't, Freeze > Thunder 2/4 > Fire 2 will get you all the ticks needed to reach max MP and does more damage than if you replaced Freeze with Blizz 2. So there's no need to reach UI 3 from UI1 in AoE. For Blizzard 3, well, the way I'm rewriting the feature would let you do whatever you want, really, since Blizzard 1 will basically be deleted from your bar at higher levels (a Balance-approved idea). So the Blizzard 2 feature and Fire 3 feature can likely be left alone, but Blizzard 3 being changed to a level sync is likely a good option.

**Comment by Grammernatzi** [comment]
I can see it being useful as an option, at the very least. Though I would personally keep it synced for just the lower levels.

**Comment by daemitus** [comment]
options added
